components/client/src/main/java/com/hotels/styx/client/connectionpool/SimpleConnectionPool.java

```diff
@@ -223,6 +229,11 @@ public void close() {
 
     @VisibleForTesting
     private class ConnectionPoolStats implements Stats {
+
+        final Histogram timeToFirstByteHistogram = new Histogram(new SlidingWindowReservoir(50));
+        private final Supplier<Long> ttfbSupplier = () -> (long) timeToFirstByteHistogram.getSnapshot().getMean();
+        private final Supplier<Long> memoizedTtfbSupplier = memoizeWithExpiration(ttfbSupplier::get, 5, MILLISECONDS)::get;
+
```

Comment:
As a documented feature, we need to expose TTFB metric as a histogram. But the supplier here just gets the mean of the recorded values.

For TTFB to be useful as a documented feature, we need to report a full histogram. The mean alone does't really say anything. Histogram on the other hand shows accurately when and how the latencies are affected.